src/common/components/session-overlay/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { signOut } from '../../actions/auth-store';
+
+import Notification from '../vanilla-modules/notification';
+
+import style from './sessionOverlay.css';
+import containerStyle from '../../containers/container.css';
+
+export class SessionOverlay extends Component {
+
+  onReloadClick(event) {
+    event.preventDefault();
+    this.props.clearSession();
+    window.location.reload();
```

Comment:
Don't we want to redirect to `/auth/authenticate`? It's a bit annoying that the session expires then you have to click through from the homepage again.

Reply:
@Lukewh We could, but (because of current sign in implementation) this would mean that users would always end up on 'My repos' page. Currently going through GH sign in always redirects to 'My repos' regardless where the sign in originated.

If we just reload (not force user to sign in automatically) we (in theory) stay on the same page. What really happens depends on the page.
If user is on any page that is available without sign in (like repo page or build page) they will stay on the same page (just not being signed in).
If user is on page only available to authenticated users (like 'My repos' or 'Add repos') they will get redirected to home page (because they are not signed in).

So the question here is more from UX perspective - once the session expires do we want to just refresh the view to properly sign out the user, or do we want to automatically sign them back in (once they click on 'Refresh') - which will be almost invisible process assuming they are still signed in to their GH account.

Reply:
I think we should aim to have something smarter happen (like use the current path as the redirect after sign in?) but for now, let's get this merged this is already way better then what we have!

Reply:
Sure. Ideally sign in should lead back where it was originated (unless it's a home page that should lead to 'My snaps' when signed in).